File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/movie_list/MoviesAdapter.java
package com.example.lifecyclesandbox.movie_list;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.lifecyclesandbox.R;

import java.util.ArrayList;

public class MoviesAdapter extends RecyclerView.Adapter<MoviesAdapter.ViewHolder> {

    private ArrayList<Movie> movieList;

    public MoviesAdapter(ArrayList<Movie> movieList){
        this.movieList = movieList;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private TextView movieTitleText;
        // TODO: Add fields on both ViewHolder of CustomAdapter, and Movie (model).

        public ViewHolder(final View view) {
            super(view);
            movieTitleText = view.findViewById(R.id.movie_title_textView);
        }
    }

    @NonNull
    @Override
    public MoviesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.movie_list_item, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MoviesAdapter.ViewHolder holder, int position) {
        String movieTitle = movieList.get(position).getMovieTitle();
        holder.movieTitleText.setText(movieTitle);
    }

    @Override
    public int getItemCount() {
        return movieList.size();
    }
}

File: LifeCycleSandBox/app/src/main/java/com/example/lifecyclesandbox/movie_list/MoviesAdapter.kt
package com.example.lifecyclesandbox.movie_list

import android.net.Uri
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import com.example.lifecyclesandbox.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.lifecyclesandbox.movie_list.model.Movie
import com.squareup.picasso.Picasso

class MoviesAdapter(private val movieList: MutableList<Movie>) : RecyclerView.Adapter<MoviesAdapter.ViewHolder>() {

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val movieTitleText: TextView = view.findViewById(R.id.movie_title_textView)
        val movieOverviewText: TextView = view.findViewById(R.id.movie_overview_textView)
        val moviePoster: ImageView = view.findViewById(R.id.movie_poster_imageView)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.movie_list_item, parent, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val movieTitle = movieList[position].title
        val movieOverview = movieList[position].overview

        val moviePosterPath = "https://image.tmdb.org/t/p/original/" + movieList[position].posterPath
        // val moviePosterPath = "https://images.unsplash.com/photo-1616530940355-351fabd9524b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1935&q=80"

        holder.movieTitleText.text = movieTitle
        holder.movieOverviewText.text = movieOverview
        Picasso.get().load(moviePosterPath).into(holder.moviePoster)
    }

    override fun getItemCount(): Int {
        return movieList.size
    }

    fun updateMovieList(popularMovieList: List<Movie>) {
        movieList.clear()
        movieList.addAll(popularMovieList)
        notifyDataSetChanged()
    }
}